libcodechecker/analyze/analysis_manager.py

```diff
@@ -77,6 +80,50 @@ def worker_result_handler(results, metadata, output_path):
 
     metadata['result_source_files'] = source_map
 
+
+def create_dependencies(action):
+    """
+    Transforms the given original build 'command' to a command that, when
+    executed, is able to generate a dependency list.
+    """
+    if 'CC_LOGGER_GCC_LIKE' not in os.environ:
+        os.environ['CC_LOGGER_GCC_LIKE'] = 'gcc:g++:clang:clang++:cc:c++'
```

Comment:
This is the only place where this is set within CodeChecker. We add this because `-E -M` and the other flags are only applicable to `gcc` and `clang`. If the user uses a custom compiler, they are expected to set this variable (otherwise `CodeChecker log` won't work either).